signer/tests/integration/postgres.rs

```diff
@@ -1268,6 +1271,89 @@ async fn we_can_fetch_bitcoin_txs_from_db() {
     signer::testing::storage::drop_db(pg_store).await;
 }
 
+/// Check that `is_signer_script_pub_key` correctly returns whether a
+/// scriptPubKey value exists in the dkg_shares table.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn is_signer_script_pub_key_checks_dkg_shares_for_script_pubkeys() {
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+    let mem = storage::in_memory::Store::new_shared();
+
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    // Okay let's put a row in the dkg_shares table.
+    let aggregate_key: PublicKey = fake::Faker.fake_with_rng(&mut rng);
+    let script_pubkey: ScriptPubKey = aggregate_key.signers_script_pubkey().into();
+    let shares = EncryptedDkgShares {
+        script_pubkey: script_pubkey.clone(),
+        tweaked_aggregate_key: aggregate_key.signers_tweaked_pubkey().unwrap(),
+        encrypted_private_shares: Vec::new(),
+        public_shares: Vec::new(),
+        aggregate_key,
+    };
+    db.write_encrypted_dkg_shares(&shares).await.unwrap();
+    mem.write_encrypted_dkg_shares(&shares).await.unwrap();
+
+    // Now we have a row in their with our scriptPubKey, let's make sure
+    // that the query accurately reports that.
+    assert!(db.is_signer_script_pub_key(&script_pubkey).await.unwrap());
+    assert!(mem.is_signer_script_pub_key(&script_pubkey).await.unwrap());
+
+    // Now we try the case where it is the script pub key is missing from
+    // the database by generating a new one (well it's unlikely to be
+    // there).
+    let aggregate_key: PublicKey = fake::Faker.fake_with_rng(&mut rng);
+    let script_pubkey: ScriptPubKey = aggregate_key.signers_script_pubkey().into();
+
+    assert!(!db.is_signer_script_pub_key(&script_pubkey).await.unwrap());
+    assert!(!mem.is_signer_script_pub_key(&script_pubkey).await.unwrap());
+
+    signer::testing::storage::drop_db(db).await;
+}
+
+/// The [`DbRead::get_signers_script_pubkeys`] function is only supposed to
+/// fetch the last 365 days worth of scriptPubKeys, but if there are no new
+/// encrypted shares in the database in a year, we should still return the
+/// most recent one.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn get_signers_script_pubkeys_returns_non_empty_vec_old_rows() {
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    let shares: model::EncryptedDkgShares = fake::Faker.fake_with_rng(&mut rng);
+
+    sqlx::query(
+        r#"
+        INSERT INTO sbtc_signer.dkg_shares (
+            aggregate_key
+            , tweaked_aggregate_key
```

Comment:
Is `tweaked` an official term here? :laughing: (no but more seriously, what does `tweaked` mean here?)

Reply:
lol, yeah it is an official term (it's used in the official taproot BIPs, like [BIP-341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki), where they detail the computations). It means that we've modified a number in such a way where we can still recover the original number if we know the modification. So in this case, we have a public key (the aggregate key) which was modified by adding some other number (which here is some hash of something or other).

I also found this simple definition from an [ipython notebook](https://github.com/bitcoinops/taproot-workshop/blob/master/2.2-taptweak.ipynb):
> Tweaking a public key means to alter it with a value (the tweak) such that it remains spendable with knowledge of the original private key and tweak.

Reply:
Interesting, I had no idea! Thx